pkg/graveler/graveler.go

```diff
@@ -1642,16 +1686,6 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 			commit.Parents = CommitParents{branch.CommitID}
 		}
 
-		if params.SourceMetaRange != nil {
```

Comment:
You have to check it here too, checks outside the lock are optimizations at best.

Reply:
You also need to know if the staging area is empty for "regular" commit - since we don't allow empty commits by default. 
(If an underlying layer is taking care of that, add a comment where it's done)

Reply:
done

Reply:
graveler.go 1724

Reply:
Not done for `params.SourceMetaRange != nil`